R/fragments-calculate2.R

```diff
@@ -0,0 +1,324 @@
+#' @title Calculate ions produced by fragmentation, including possible modifications
+#' 
+#' @aliases calculateFragments2, modificationPositions, cumsumFragmentMasses, character,missing-method
+#'
+#' @name calculateFragments2
+#' 
+#' @param sequence character() providing a peptide sequence.
+#' 
+#' @param type character vector of target ions; possible values: 
+#' c("a", "b", "c", "x", "y", "z"). Default is type = c("b", "y").
+#' 
+#' @param z numeric with a desired charge state; default is 1.
+#' 
+#' @param fixed_modifications A named numeric vector of used fixed modifications. 
+#' The name must correspond to the one-letter-code of the modified amino acid 
+#' and the numeric value must represent the mass that should be added to the 
+#' original amino accid mass, default: Carbamidomethyl modifications = 
+#' c(C = 57.02146). Use Nterm or Cterm as names for modifications that should 
+#' be added to the amino respectively carboxyl-terminus.
+#' 
+#' @param variable_modifications A named numeric vector of variable modifications.
+#' Depending on the maximum number of modifications (`max_mods`), all possible 
+#' combinations are returned.
+#' 
+#' @param max_mods A numeric indicating the maximum number of variable modifications 
+#' allowed on the sequence at once. Does not include fixed modifications. 
+#' Default value is positive infinity.
+#' 
+#' @param neutralLoss list, it has to have two named elments, namely water and
+#' ammonia that contain a character vector which type of neutral loss should be 
+#' calculated. Currently neutral loss on the C terminal "Cterm", at the amino 
+#' acids c("D", "E", "S", "T") for "water" (shown with an ⁠_⁠) an
+#' d c("K", "N", "Q", "R") for "ammonia" (shown with an *) are supported.
+#' 
+#' @param verbose logical(1). If TRUE (default) the used modifications are printed.
+#' 
+#' @return A dataframe showing all the 
+#' ions produced by fragmentation with all possible combinations of modifications.
+#' The used variable modifications are displayed in the peptide column through the
+#' use of amino acids within brackets. Fixed modifications are not displayed. 
+#' 
+#' 
+#' @examples
+#' calculateFragments2(sequence = "ARGSHKATC", type = c("b", "y"), z = 1, 
+#' fixed_modifications = c(C = 57), variable_modifications = c(S = 79, Y = 79, T = 79),
+#' max_mods = 2)
+#' @export
+calculateFragments2 <- function(sequence, 
+                                type = c("b", "y"), 
+                                z = 1,
+                                fixed_modifications = c(C = 57.02146),
+                                variable_modifications = NULL,
```

Comment:
I am wondering if an empty `character()` wouldn't be a better fit than NULL? We should discuss the pros and cons.